### PR TITLE
Chat styling, HTML rewrite, and autoscroll

### PR DESCRIPTION
**Pull request by salnad**
## Changes Made
- Restyle chat feature (with padding, margins, etc) to make it distinguishable from the rest of the page.
- Add CSS classes for certain messages that will be used in future (eg. correct guess, 'close' notification, etc).
- Add autoscroll, so that when a new message is sent / recieved, the chat window scrolls to the bottom of the page.

Example of new message types and autoscroll in action available to watch [here](https://screencast.googleplex.com/cast/NTk2NTk2NjQyNzAyOTUwNHwyODMzOGQxNy1hZQ)

### CODE REVIEW

**Comment by salnad** [comment]
There was an error when merging master into my current branch due to the pom.xml, so last commit was to fix that.

File: src/main/webapp/chat.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#chat {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  align-items: center;
```
Comment (guavafanclub) [inline src/main/webapp/chat.css:5]: Can you alphabetize declarations? See section 4.2.1 https://google.github.io/styleguide/htmlcssguide.html

File: src/main/webapp/chat.html [hunk 2]
```diff
@@ -2,11 +2,20 @@
 <head>
   <title>Pusher Test</title>
   <script src="https://js.pusher.com/6.0/pusher.min.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.js"></script>
   <script src="chat.js"></script>
+  <link rel="stylesheet" type="text/css" href="chat.css" />
```
Comment (guavafanclub) [inline src/main/webapp/chat.html:7]: Can you rename the stylesheet styles.css, since we hope to use one across pages?

File: src/main/webapp/chat.html [hunk 3]
```diff
@@ -2,11 +2,20 @@
...
-  <input type="text" id="chat-input" placeholder="Type message here..."></input>
-  <button onclick="addToChat()">click</button>
+  <div id="chat">
+    <div id="chatbox"></div>
+    <div id="chat-input">
+      <input
+        type="text"
+        id="chat-input-box"
+        placeholder="Type message here..."
+      />
```
Comment (guavafanclub) [inline src/main/webapp/chat.html:17]: move this "/>" to line 16.  Indent at least 4 spaces. See section 3.2.2 https://google.github.io/styleguide/htmlcssguide.html

File: src/main/webapp/chat.js [hunk 4]
```diff
@@ -19,10 +19,12 @@ async function addToChat() {
 }
 
 function updateChat(data) {
-  let newMessage = document.createElement("li");
-  newMessage.appendChild(document.createTextNode(data.message));
-  let chat = document.getElementById("chat");
-  chat.appendChild(newMessage);
+  let message = data.message;
+  let newChatItem = `<p class="message"><span class="username">User: </span>${message}</p>`;
```
Comment (pruess-google) [inline src/main/webapp/chat.js:23]: Is the plan to eventually replace "User" with the username?

Comment (salnad) [reply src/main/webapp/chat.js:23]: Yeah, should see that in next PR

File: src/main/webapp/chat.js [hunk 5]
```diff
@@ -19,10 +19,12 @@ async function addToChat() {
...
+  $("#chatbox").append(newChatItem);
```
Comment (alekseychaikovsky) [inline src/main/webapp/chat.js:24]: the chat item itself doesn't seem like too complicated to construct using vanilla JS.

Neither does the append method.

Did you try doing this before importing the entire Jquery Library?

File: src/main/webapp/style.css [hunk 6]
```diff
@@ -0,0 +1,35 @@
+#chat {
+  align-items: center;
+  border-width: 1px;
+  border-style: solid;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  padding: 10px 10px;
```
Comment (alekseychaikovsky) [inline src/main/webapp/style.css:8]: could shorten this to 10px. Up to you to change or not, not a big deal

File: src/main/webapp/style.css [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  padding: 5px 10px;
+  width: 300px;
+}
+
+.message-anouncement {
+  color: #b34700;
+}
+
+.message-correct {
+  color: green;
```
Comment (alekseychaikovsky) [inline src/main/webapp/style.css:26]: stick to using the hex code.

Also, since this is now the "global" css, its probably worth defining these colors as variables to better describe them, and then referring to the variables in these classes.

This will also help the app look consistent when other contributors are styling their parts.

Check out this documentation for how to do this:
https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties
